content/upcoming/announcing_babelfish.md

```diff
@@ -0,0 +1,82 @@
+---
+author: abeaumont
+date: 2017-06-06
+title: "Announcing Babelfish"
+draft: true
+image: **TODO: Babelfish image (/post/announcing-babelfish/babelfish.jpg)**
+description: "Announcing Babelfish, the project we are developing to build
+representations of source code."
+categories: ["technical"]
+---
+
+At source{d} we believe there's a possibility for programs to write their own
+code and we are exploring that possibility. Our
+[Roadmap](https://blog.sourced.tech/post/our-roadmap/) states the first step in this
+direction as: *Build representations of source code, developers and projects.*
+
+Today we are announcing Babelfish, the project we are developing to build this
+representations of source code.
+
+## What's Babelfish?
+
+Babelfish is a universal code parser. It can parse any file, in any language,
+extract an abstract syntax tree (AST) from it, and convert it to a Universal
+Abstract Syntax Tree [(UAST)](https://doc.bblf.sh/uast/specification.html)
+
+Well, that's our objective, we are not there just yet. We have designed an
+architecture we're currently working on to achieve that.
+
+## The architecture of Babelfish
+
+**TODO: Architecture image**
+
+As shown in the figure, Babelfish is a client/server application. A client is a
+language analysis tool that wants to perform some analysis on a source code. It
+connects to the server handle source code parsing by using the needed language
+driver containers.
+
+Language drivers are divided in two parts, in order to minimise the work of
+developing a new driver:
+
+- A language code parser, which builds a native AST. This parser can be built
+  directly from target language's compiler tools or libraries.
+- An AST normalizer written in Go, that gives the UAST as a result.
+
+You can look at the [documentation](https://doc.bblf.sh/architecture.html) for
```

Comment:
It would be good to add a section or paragraph about the fact that Babelfish server uses containers to be able to run drivers using disparate ecosystems with minimal overhead for the user. Make sure you include the words `libcontainer` and `Docker` in the blog post ;)